### ISSUE TITLE
[API/RPC] Use one uniform naming format

### ISSUE THREAD

**Issue by SebastianJ** [harmony-one/harmony#3013]
## Summary

The current hmy API:s both v1/v2 does not use a uniform naming scheme for the returned RPC fields.

Some fields are returned as snake_case, some as camel_case, some using dashes between words and some fields starting with capitalized letters.

This has messed up e.g. the testing framework more than once - and for some fields the standard has been changed haphazardly between using underscores and dashes to separate fields back and forth multiple times.

If we expect people to use the API they should be able to rely on RPC fields following a certain standard and that the standard isn't changed willy-nilly.

Here's for example a definition where three different standards (camelCase, snake_case, Capitalized) are used:
https://github.com/harmony-one/harmony/blob/master/internal/hmyapi/apiv2/types.go#L71-L92

The inconsistency between the naming and changing it back and forth between releases is also the reason for e.g. why the D1/U1 test cases failed for some people on Dryrun (https://github.com/harmony-one/harmony/issues/2995), but didn't fail for others. It came down to when the tests were running, and what agains version of the API (it switched between encoding delegator addresses as `delegator-address` and `delegator_address`.

**Comment by stephen-tse** [comment]
i'd recommend snake_case, or strictly follow ethereum's convention

**Comment by Daniel-VDM** [comment]
I agree that this needs to be standardized, and would prefer underscores. However, my concern is backward compatibility with existing tools. We have to consider breaking backward compatibility for the sake of convention or add another version of the RPCs. The different versions of the RPCs could be cleaned up at the RPC layer, so we could add another version of the RPCs that would be standardized while doing the cleanup.
